draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -790,57 +793,47 @@ The transparency service maintains a mirror of the CA's latest batch number, and
 
-4. If `new_latest_batch` is less than `old_latest_batch`, abort this procedure with an error.
+How the relying party does this has no direct impact on certificate issuance ({{issuing-certificates}}) or usage ({{using}}). Different relying parties can obtain validity windows from different sources or apply different policies, while supporting the same unmodified certificates and CAs. Thus this section does not prescribe particular policies or mechanisms, but provides examples and general guidance. Relying parties MAY implement policies other than those described below, and MAY incorporate entities acting in roles not described in this document.
 
-5. If the issuance time for batch `new_latest_batch` is after the current time (see {{parameters}}), abort this procedure with an error.
+Relying parties SHOULD ensure transparency by obtaining validity windows from the CA and/or some combination of trusted mirrors. The relying party picks sources which, together, are trusted to satisfy the requirements described in {{mirrors}}. Mirrors allow the relying party to maintain transparency in the face of a misbehaving CA that may, for example, stop serving some unauthorized certificate in a batch to evade detection.
 
-6. Fetch the validity window with `new_latest_batch` from each mirror that returned an equal or higher latest batch number. If any fetches fail, or if the results do not match across all mirrors, abort this procedure with an error.
+Some relying parties regularly contact a trusted update service, either for software updates or to update individual components, such as the services described in {{CHROMIUM}} and {{FIREFOX}}. Where these services are already trusted for the components such as the trust anchor list or certificate validation software, that service could be used as a single trusted mirror.
 
-7. Verify the validity window signature, as described in {{signing}}. If the signature is invalid, abort this procedure with an error.
+A relying party might trust a combination of mirrors, and only accept a validity window once some minimum number of mirrors have consumed it. Such a relying party would fetch the latest batch number from each mirror, then compute the highest batch number that is contained in sufficiently many mirrors. It would then fetch that validity window from each containing mirror, check their tree hashes match and, if so, accept the update.
 
-8. If the old and new validity windows contain overlapping batch numbers, verify that the tree hashes match. If not, abort this procedure with an error.
+This procedure MAY also be performed by the relying party's update service, or some aggregation service, which forwards the signatures from each mirror to the relying party. This allows the relying party to avoid directly contacting each mirror. [[TODO: This assumes mirrors re-sign validity windows, but we have not actually defined that yet. #101]]
```

Comment:
Just noting that if we go this route, it would be nice to leverage the co-signing mechanism of [tlog-witness](https://github.com/C2SP/C2SP/blob/main/tlog-witness.md).